backend/src/main/java/ai/verta/modeldb/artifactStore/storageservice/s3/S3Service.java

```diff
@@ -137,7 +138,25 @@ private void fetchCredentialsAndInitializeS3ClientWithTemporaryCredentials(Regio
     }
   }
 
-  private void initializeS3ClientWithTemporaryCredentials(Regions clientRegion) {
+  private synchronized void initializeS3ClientWithTemporaryCredentials(
+      Regions clientRegion, AmazonServiceException ex) {
+
+    if (ex == null && s3Client != null) {
```

Comment:
comment for conrado's review: before we initializethe s3 client we check if we already have a valid s3 client